From a4d4023e24080f33acc12d4871756731b9c406ce Mon Sep 17 00:00:00 2001
From: Tomas Zak <tomas.zak@turris.com>
Date: Thu, 15 Jan 2026 12:31:22 +0100
Subject: [PATCH] uboot-tools: fix SWIG compilation error

Since SWIG version 4.3 has changed language specific AppendOutput functions.
See:
https://swig.org/Release/CHANGES
https://lists.infradead.org/pipermail/lede-commits/2024-December/023727.html
---
 .../patches/003-SWIG-python-fix.patch         | 51 +++++++++++++++++++
 1 file changed, 51 insertions(+)
 create mode 100644 package/boot/uboot-tools/patches/003-SWIG-python-fix.patch

diff --git a/package/boot/uboot-tools/patches/003-SWIG-python-fix.patch b/package/boot/uboot-tools/patches/003-SWIG-python-fix.patch
new file mode 100644
index 0000000000..6d826c621a
--- /dev/null
+++ b/package/boot/uboot-tools/patches/003-SWIG-python-fix.patch
@@ -0,0 +1,51 @@
+From 9991114ff2be700db411603a21cf5829c1885354 Mon Sep 17 00:00:00 2001
+From: Markus Volk <f_l_k@t-online.de>
+Date: Mon, 28 Oct 2024 18:26:45 +0100
+Subject: [PATCH] pylibfdt/libfdt.i: Use SWIG_AppendOutput
+MIME-Version: 1.0
+Content-Type: text/plain; charset=UTF-8
+Content-Transfer-Encoding: 8bit
+
+Swig has changed language specific AppendOutput functions. The helper
+macro SWIG_AppendOutput remains unchanged. Use that instead
+of SWIG_Python_AppendOutput, which would require an extra parameter
+since swig 4.3.0.
+
+| /home/flk/poky/build-hypr/tmp/work/x86_64-linux/python3-dtc-native/1.7.0/git/pylibfdt/libfdt_wrap.c: In function ‘_wrap_fdt_next_node’:
+| /home/flk/poky/build-hypr/tmp/work/x86_64-linux/python3-dtc-native/1.7.0/git/pylibfdt/libfdt_wrap.c:5598:17: error: too few arguments to function ‘SWIG_Python_AppendOutput’
+|  5598 |     resultobj = SWIG_Python_AppendOutput(resultobj, val);
+
+Signed-off-by: Markus Volk <f_l_k@t-online.de>
+---
+ pylibfdt/libfdt.i | 6 +++---
+ 1 file changed, 3 insertions(+), 3 deletions(-)
+
+--- a/scripts/dtc/pylibfdt/libfdt.i_shipped
++++ b/scripts/dtc/pylibfdt/libfdt.i_shipped
+@@ -1037,7 +1037,7 @@ typedef uint32_t fdt32_t;
+ 			fdt_string(fdt1, fdt32_to_cpu($1->nameoff)));
+ 		buff = PyByteArray_FromStringAndSize(
+ 			(const char *)($1 + 1), fdt32_to_cpu($1->len));
+-		resultobj = SWIG_Python_AppendOutput(resultobj, buff);
++		resultobj = SWIG_AppendOutput(resultobj, buff);
+ 	}
+ }
+ 
+@@ -1076,7 +1076,7 @@ typedef uint32_t fdt32_t;
+ 
+ %typemap(argout) int *depth {
+         PyObject *val = Py_BuildValue("i", *arg$argnum);
+-        resultobj = SWIG_Python_AppendOutput(resultobj, val);
++        resultobj = SWIG_AppendOutput(resultobj, val);
+ }
+ 
+ %apply int *depth { int *depth };
+@@ -1092,7 +1092,7 @@ typedef uint32_t fdt32_t;
+            if (PyTuple_GET_SIZE(resultobj) == 0)
+               resultobj = val;
+            else
+-              resultobj = SWIG_Python_AppendOutput(resultobj, val);
++              resultobj = SWIG_AppendOutput(resultobj, val);
+         }
+ }
+ 
-- 
2.52.0

